packages/dnb-eufemia/src/components/date-picker/hooks/usePartialDates.ts

```diff
@@ -0,0 +1,39 @@
+import { useRef } from 'react'
+
+export type PartialDates = {
+  partialDate?: string
+  partialStartDate?: string
+  partialEndDate?: string
+}
+
+export default function usePartialDates() {
+  const partialDatesRef = useRef<PartialDates>({
+    partialStartDate: null,
+    partialEndDate: null,
+  })
+
+  function setPartialDates(partialDates: PartialDates) {
```

Comment:
```suggestion
  const setPartialDates = useCallback((partialDates: PartialDates) => {
  etc.
```

Using a function without useCallback will create a new instance and can create issues when used in other hooks later.